### PR TITLE
@kanaabe => Upgrade to React 16 + Enzyme 3

### PR DESCRIPTION
**Pull request by damassi**
This updates minor semver packages via `yarn upgrade-interactive`, updates React to 16 and Enzyme to 3. For enzyme in particular, read over the [migration guide](https://github.com/airbnb/enzyme/blob/master/docs/guides/migration-from-2-to-3.md) as there are a handful of breaking changes and ([this issue](https://github.com/airbnb/enzyme/issues/1201) might also be good to read. 

There were a few places where I've added `FIXME TEST:` as I couldn't quite figure out how to fix, but should be quick to go over once merged.

### CODE REVIEW

File: client/assets/main.coffee [hunk 1]
```diff
@@ -7,6 +7,14 @@ Backbone = require 'backbone'
 window.jQuery = window.$ = $ = require 'jquery'
 window.global = window
 
+React = require 'react'
+DOM = require 'react-dom-factories'
+createClass = require('create-react-class')
+
+# Patch React 16 with deprecated helpers. Can remove once CoffeeScript goes away
+React.DOM = DOM
+React.createClass = createClass
```
Comment (kanaabe) [inline client/assets/main.coffee]: Oh nice, we were wondering about how we would do this :)

Comment (damassi) [reply client/assets/main.coffee]: ❤️ JavaScript

**Comment by kanaabe** [comment]
Wow @damassi this is legit 👍 . Once the tests pass (and feel free to continue pending things if it becomes too much of a pain), let's merge and make sure staging looks great. I created a card so we don't forget to address them.

**Comment by damassi** [comment]
Alright! this should be ready to go

**Comment by damassi** [comment]
All of the Enzyme 3 changes were tough! make sure to check out the migration guide and let me know when y'all have checked out the `FIXME TESTS` issue (out of curiousity)

**Pull request merged**